nomad/structs/services.go

```diff
@@ -272,6 +272,7 @@ func (sc *ServiceCheck) Hash(serviceID string) string {
 	io.WriteString(h, sc.Interval.String())
 	io.WriteString(h, sc.Timeout.String())
 	io.WriteString(h, sc.Method)
+	io.WriteString(h, sc.TaskName)
```

Comment:
This should not be necessary as `serviceID` includes the task name.

Reply:
Will fix.